modules/renter/contractor/contractor.go

```diff
@@ -237,13 +242,14 @@ func NewCustomContractor(cs consensusSet, w wallet, tp transactionPool, hdb host
 
 		interruptMaintenance: make(chan struct{}),
 
-		staticContracts: contractSet,
-		downloaders:     make(map[types.FileContractID]*hostDownloader),
-		editors:         make(map[types.FileContractID]*hostEditor),
-		oldContracts:    make(map[types.FileContractID]modules.RenterContract),
-		renewedIDs:      make(map[types.FileContractID]types.FileContractID),
-		renewing:        make(map[types.FileContractID]bool),
-		revising:        make(map[types.FileContractID]bool),
+		staticContracts:     contractSet,
+		downloaders:         make(map[types.FileContractID]*hostDownloader),
+		editors:             make(map[types.FileContractID]*hostEditor),
+		oldContracts:        make(map[types.FileContractID]modules.RenterContract),
+		contractIDToPubKey:  make(map[types.FileContractID]types.SiaPublicKey),
+		pubKeysToContractID: make(map[string]types.FileContractID),
```

Comment:
It looks like this map currently only stores the most recent contract associated with a host pubkey?

Reply:
Yes, that's intentional. The contractor only needs this map to return the most recent contract with a host to the caller of the interface. Other modules don't really care about older revisions.